client/app/routes/collection/add.js

```diff
@@ -1,4 +1,124 @@
+                },
+                preprint_title_widget: {
+                    state: ['undefined'],
+                    value: undefined
+                },
+                save_upload_section_widget: {
+                    state: ['undefined'],
+                    value: undefined
+                }
+            },
+            actions: [{
+                type: 'create_widget',
+                parameters: {
+                    widget_component: 'file-uploader',
+                    description: 'Choose the preprint file to upload',
+                    section: 'upload',
+                    output: 'preprint_file_data'
+                },
+                output: 'preprint_file_upload_widget',
+                conditions: [{
+                    all: [{
+                        parameter: 'upload_section',
+                        state: 'unsaved',
+                    }, {
+                        parameter: 'preprint_file_upload_widget',
+                        state: 'undefined'
+                    }]
+                }]
+            }, {
+                type: 'create_widget',
+                parameters: {
+                    widget_component: 'text-field',
+                    description: 'Enter the title for this preprint',
+                    section: 'upload',
+                    output: 'preprint_file_name'
+                },
+                output: 'preprint_title_widget',
+                conditions: [{
+                    all: [{
+                        parameter: 'preprint_title_widget',
+                        state: 'undefined',
+                    }, {
+                        parameter: 'preprint_file_data',
+                        state: 'defined',
+                    }, {
+                        parameter: 'upload_section',
+                        state: 'editing',
+                    }],
+                }]
+            }, {
+                type: 'create_widget',
+                parameters: {
+                    widget_component: 'button-widget',
+                    description: 'Save this section',
+                    section: 'upload',
+                    output: 'upload_section'
+                },
+                output: 'save_upload_section_widget',
+                conditions: [{
+                    all: [{
+                        parameter: 'save_upload_section_widget',
+                        state: 'undefined',
+                    }, {
+                        parameter: 'preprint_file_data',
+                        state: 'defined',
+                    }, {
+                        parameter: 'preprint_file_name',
+                        state: 'defined'
+                    }, {
+                        parameter: 'upload_section',
+                        state: 'editing',
+                    }],
+                }]
+            }]
+        };
+    },
+
+    setupController(controller, model) {
+        controller.set('form_config', model);
```

Comment:
No, though I figured it'd be convenient to have the whole data structure somewhere easily available if we need it.